src/openrct2/core/FileStream.cpp

```diff
@@ -98,9 +100,12 @@ namespace OpenRCT2
             throw IOException(String::StdFormat("Unable to open '%s'", path));
         }
 
-        Seek(0, STREAM_SEEK_END);
-        _fileSize = GetPosition();
-        Seek(0, STREAM_SEEK_BEGIN);
+#ifdef _WIN32
+        _fileSize = _filelengthi64(_fileno(_file));
+#else
+        std::error_code ec;
+        _fileSize = fs::file_size(fs::u8path(path), ec);
+#endif
```

Comment:
Why not use `Platform::GetSize` here? Maybe I missed something.

In any case, I'm (slowly) trying to keep all OS ifdefs on the platform files, for an easier to read code overall, so I'd be very interested in making this a bit friendlier.

Reply:
I could use that, but in this case on Windows I could get the size directly from a `FILE*` instead of a path.